In [1]:
import os
import numpy as np
import pylab as pl
import json
from scipy.optimize import curve_fit,minimize
c=json.load(open('pui.json'))
pl.rcParams.update(c)

%pylab inline

Populating the interactive namespace from numpy and matplotlib


C:\Users\ManushiM\Anaconda\lib\site-packages\matplotlib\__init__.py:855: UserWarning: svg.embed_char_paths is deprecated and replaced with svg.fonttype; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))


In [2]:
alpha = 0.05

P_0 = 52.1 * 0.01
P_1 = 53.3 * 0.01
n_0 = 409
n_1 = 564

nt_0 = P_0 * n_0
nt_1 = P_1 * n_1

In [3]:
#define the sample proportion first
sp=(P_0 * n_0 + P_1 * n_1)/(n_1 + n_0)
print sp

0.527955806783


In [4]:
#calculation of standard deviation by a one-line function

sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((nt_0 + nt_1)/(n_0 + n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.521 409 564 0.0324224126866


In [5]:
#calculation of z-score

zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y

0.370114343926


Based on the Z-score table,

In [6]:
p_2y = 1 - 0.6443

In [7]:

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


### For the chi-squared test

In [8]:
Ntot = 973
expected = 564*409*115.787*857.213
sample_values = [[14.7*5.64, 85.3*5.64], [11.9*4.09, 88.1*4.09]]
chi_stats = lambda N, values, expect: N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chi_stats(Ntot, sample_values, expected)

1.772893977


3.84 is the chi-squared test statistic for 1 degree of freedom at alpha = 0.05
We have, chi-squared test vale = 1.7729, degree of freedom = 1, alpha = 0.05
Also our p value > alpha
Hence, we cannot reject the null hypothesis.